sanic/signals.py

```diff
@@ -98,14 +146,23 @@ async def dispatch(
         *,
         context: Optional[Dict[str, Any]] = None,
         condition: Optional[Dict[str, str]] = None,
-    ) -> asyncio.Task:
-        task = self.ctx.loop.create_task(
-            self._dispatch(
-                event,
-                context=context,
-                condition=condition,
-            )
+        fail_not_found: bool = True,
+        inline: bool = False,
+        reverse: bool = False,
+    ) -> Union[asyncio.Task, Any]:
+        dispatch = self._dispatch(
+            event,
+            context=context,
+            condition=condition,
+            fail_not_found=fail_not_found and inline,
+            reverse=reverse,
         )
+        logger.debug(f"Dispatching signal: {event}")
+
+        if inline:
+            return await dispatch
+
+        task = asyncio.get_running_loop().create_task(dispatch)
         await asyncio.sleep(0)
```

Comment:
any particular reason for this `asyncio.sleep` call?

Reply:
I am not in love with it, and open to other solutions. I should probably add a comment to explain. The reason it is there is to yield back to the loop one time which is useful in avoiding a deadlock when waiting on `await app.event`.

Reply:
well, have you bumped into any deadlocks without it?

Reply:
Yes.

Reply:
I think the case in the docs will deadlock without it:

```python
async def wait_for_event(app):
    while True:
        print("> waiting")
        await app.event("foo.bar.baz")
        print("> event found\n")

@app.after_server_start
async def after_server_start(app, loop):
    app.add_task(wait_for_event(app))

```